CRM/Utils/SQL/BaseParamQuery.php

```diff
@@ -202,6 +203,7 @@ public function offsetExists($offset) {
    * @see param()
    * @see ArrayAccess::offsetGet
    */
+  #[ReturnTypeWillChange]
```

Comment:
does changing `   * @return mixed` to    `* @return null|int` make this unnecessary?

Reply:
so i think what it wants is a function hint of `:mixed`  but that has different connotations in php7.x to php8.x

Reply:
Yeah I don't think there's any other way to do this to support both php 7 and 8, since mixed doesn't exist in php 7 (it thinks it's a class type and so then wants an instance of class `mixed`).

Reply:
One thing to think about, but maybe there's no way around it, is that it won't be possible to support both php 7 and 9 at the same time, because php 9 will ignore these `#[ReturnTypeWillChange]` and _require_ a return type specified, but php 7 won't let you specify mixed. One possibility, but is a lot of work, is to refactor all these mixed functions to not be so vague. But I don't think civi will be the only project in that boat, so hopefully php 9 will be far enough out that civi will be able to stop supporting php 7 by then. It just might cause difficulty while trying to prep for php 9.